src/bridge/cockpitpackages.c

```diff
@@ -1044,7 +991,15 @@ handle_packages (CockpitWebServer *server,
     {
       cockpit_web_response_set_cache_type (response, COCKPIT_WEB_RESPONSE_NO_CACHE);
     }
-  package_content (packages, response, name, path, languages[0], out_headers);
+
+  protocol = g_hash_table_lookup (headers, "X-Forwarded-Proto");
```

Comment:
in cockpitwebservice, this is configurable as ("WebService", "ProtocolHeader"). Mostly because different webservers have different defaults for the header name.